### ISSUE TITLE
Headers getting removed for Microsoft Graph API in iPad browsers

### ISSUE THREAD

**Issue by SnehalJ1509** [angular/angular#40279]
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/common/http

### Is this a regression?

 I am not sure about this as this is the first time we are implementing it with the below mentioned setup

### Description

We have an Ionic/Angular PWA app with Microsoft Azure login. For login we have used [@azure/msal-angular](https://www.npmjs.com/package/@azure/msal-angular) v1.0.0 plugin. After login we fetch 2 access tokens and store it in localStorage. One access token for normal app API and another for Microsoft Graph API where we fetch a members list from Azure Active Directory Group and use it in the app. We are using HttpClient from @angular/common/http package to make these API calls.

All the API calls works fine for desktop browsers but the problem is while sending the Graph API call https://graph.microsoft.com/v1.0/groups/group-id/members from iPad browsers(tested in Safari, Chrome and Firefox), the headers are getting removed, so the access token is not passed resulting in 401 Unauthorized error. **This issue is happening exclusively for Graph API call in iPad browsers**. Kindly refer below screenshots.

_**Normal API call with access token in headers**_
![Screen Shot 2020-12-30 at 12 56 29 PM](https://user-images.githubusercontent.com/13345195/103348672-bb363400-4ac0-11eb-99ea-48d7786ee414.png)

**_Graph API call with no headers_**
![Screen Shot 2020-12-30 at 12 56 53 PM](https://user-images.githubusercontent.com/13345195/103348864-49aab580-4ac1-11eb-82f4-667441c62e35.png)

**_Resulting in 401 Unauthorized error_**
![Screen Shot 2020-12-30 at 12 57 45 PM](https://user-images.githubusercontent.com/13345195/103348977-9a221300-4ac1-11eb-881b-10a08c5d1a8b.png)

## 🔬 Minimal Reproduction

 Due to the app architecture it will be difficult to share minimal reproduction of the issue but I'll try if possible and update the stackblitz or repo URL here.

## 🔥 Exception or Error
<pre><code>

401 Unauthorized error
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>

Angular CLI: 9.1.11
Node: 10.15.1
OS: darwin x64

Angular: 9.1.11
... cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.901.8
@angular-devkit/build-angular     0.901.8
@angular-devkit/build-optimizer   0.901.8
@angular-devkit/build-webpack     0.901.8
@angular-devkit/core              9.1.8
@angular-devkit/schematics        9.1.8
@angular/service-worker           9.1.12
@ngtools/webpack                  9.1.8
@schematics/angular               9.1.8
@schematics/update                0.901.11
rxjs                              6.5.5
typescript                        3.8.3
webpack                           4.42.0
</code></pre>

**Anything else relevant?**

Tested on Safari, Chrome and Firefox on iPad iOS 13.7 and 14.2

Other app info
    "@angular/common": "~9.1.6",
    "@angular/core": "~9.1.6",
    "@angular/forms": "~9.1.6",
    "@angular/platform-browser": "~9.1.6",
    "@angular/platform-browser-dynamic": "~9.1.6",
    "@angular/router": "~9.1.6",
    "@angular/service-worker": "~9.1.6",
    "@azure/msal-angular": "^1.0.0",
    "@ionic-native/core": "^5.0.7",
    "@ionic/angular": "^5.1.1",
    "@ionic/pwa-elements": "^3.0.0",

**Comment by jelbourn** [comment]
@SnehalJ1509 without a reproduction or some code there's not really anything we can look at here

**Comment by SnehalJ1509** [comment]
@jelbourn yes I am trying to figure out how to give a minimal reproduction. Due to the app architecture it is taking some time. Once I am able to do so, I'll share the link.

**Comment by SnehalJ1509** [comment]
Closing this as I found a solution for this issue. Actually it is more of a workaround rather than solution. 

All the API calls in our app goes through the following flow: 
Ionic Page (API call initiated with data) -> API service (API URL and data passed) -> HTTP service (HttpClient get/post call to API URL with data is made from here)

For the Graph API we tried this: 
Ionic Page (API call initiated with API URL and data) -> HttpClient get call to Graph API is made directly from Ionic Page

There is not much difference in both the flow but strangely the second one works in iPad browsers but the first one does not. There is definitely something weird happening somewhere but as this is time consuming even to give a minimal reproduction, currently we'll stick to this workaround and close this issue.
